### ISSUE TITLE
Error `primaryKey is not defined` creating ManyToMany relationship with `--incremental-changelog`

### ISSUE THREAD

**Issue by egvimo** [jhipster/generator-jhipster#15282]
##### **Overview of the issue**

I can't add a ManyToMany relationship to an existing JDL if I use incremental changelog.

```
ReferenceError: C:\Users\...\node_modules\generator-jhipster\generators\database-changelog-liquibase\templates\src\main\resources\config\liquibase\changelog\updated_entity_constraints.xml.ejs:107
    105|     _%>
    106|
 >> 107|         <addForeignKeyConstraint baseColumnNames="<%= primaryKey.fields.map(field => relationship.columnName + '_' + field.columnName).join(', ') %>"
    108|                                  baseTableName="<%= relationship.joinTable.name %>"
    109|                                  constraintName="<%= constraintName %>"
    110|                                  referencedColumnNames="<%= primaryKey.fields.map(field => field.columnName).join(', ') %>"

primaryKey is not defined
    at module.exports.eval (eval at compile (C:\Users\...\node_modules\generator-jhipster\node_modules\ejs\lib\ejs.js:662:12), <anonymous>:168:26)     
    at updated_entity_constraints.xml (C:\Users\...\node_modules\generator-jhipster\node_modules\ejs\lib\ejs.js:692:17)
    at C:\Users\...\node_modules\generator-jhipster\node_modules\ejs\lib\ejs.js:258:40
    at new Promise (<anonymous>)
    at tryHandleCache (C:\Users\...\node_modules\generator-jhipster\node_modules\ejs\lib\ejs.js:256:14)
    at Object.exports.renderFile (C:\Users\...\node_modules\generator-jhipster\node_modules\ejs\lib\ejs.js:489:10)
    at Object.renderContent (C:\Users\...\node_modules\generator-jhipster\generators\utils.js:286:23)
    at module.exports.template (C:\Users\...\node_modules\generator-jhipster\generators\generator-base-private.js:758:26)
    at C:\Users\...\node_modules\generator-jhipster\generators\generator-base.js:2222:47
    at Array.map (<anonymous>) {
  path: 'C:\\Users\\...\\node_modules\\generator-jhipster\\generators\\database-changelog-liquibase\\templates\\src\\main\\resources\\config\\liquibase\\changelog\\updated_entity_constraints.xml.ejs'
}
```

##### **Reproduce the error**

Generate the app below without the relationship, and then with the relationship:

```
jhipster import-jdl app.jdl --with-entities --incremental-changelog --skip-git --skip-install --force
```

JDL:

```
application {
    config {
        baseName sample
        packageName sample
        applicationType monolith
        authenticationType oauth2
        buildTool gradle
        clientFramework angularX
        prodDatabaseType postgresql
        devDatabaseType postgresql
        testFrameworks [cypress]
    }
    entities *
}

entity Abc {
    name String unique required
    otherField String
}

entity Xyz {
    uniqueField String unique required
    anotherField String
}

relationship ManyToMany {
    Abc{xyz(name)} to Xyz{abc(name)} // <-- This is not working
}
```

##### **JHipster Version(s)**

7.0.1

##### **JHipster configuration**

See my [example app here](https://github.com/egvimo/jhipster-sample-app/tree/15282).

- [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by egvimo** [comment]
I still have the issue.

**Comment by Tcharl** [comment]
Can you please try with 7.1.0? I had the same issue with 7.0.1

**Comment by egvimo** [comment]
Same error with 7.1.0. 😞

**Comment by egvimo** [comment]
Any progress on this?
At the moment it is not possible to create a many to many relationship with incremental changelog. ☹️

**Comment by Tcharl** [comment]
Hi,

There's an error in your jdl: the referenced attribute should exist:
correct value
```
relationship ManyToMany {
    Abc{xyz(uniqueField)} to Xyz{abc(name)} // <-- This is not working
}
```

**Comment by egvimo** [comment]
@Tcharl Unfortunately this does not solve the issue. Same `primaryKey is not defined` error. Are you not able to reproduce it?

**Comment by Tcharl** [comment]
Reproduced :-).

It works well if we generate everything starting with the relationship.

@mshima please, as you've worked on the incremental changelog any quick idea? Otherwise I'll investigate a bit on it.

**Comment by mshima** [comment]
I didn't tried to investigate why CI didn't catch the problem, but seems to fix the problem.

**Comment by egvimo** [comment]
I think I know why CI didn't catch the problem.

My initial data model was with a `OneToMany` relationship. As I changed that to `ManyToMany` the generated changelog was missing the 'new' parts. Only the 'drop' parts were being generated. So I had to generate twice. Unfortunately leading to this error.
Some time ago I had a similar case. I did multiple changes to the JDL, but the changelog was missing some parts. Maybe there is an 'if/else' where only one part of the changelog are being generated and not the other.
